### DISCUSSION TITLE
i got an error setting the ROCKET_SSL env variable on Windows

### DISCUSSION THREAD

**Discussion by beppe9000** [dani-garcia/vaultwarden#1542] [Q&A]
I built bitwarden_rs on windows. Now I want to run it in ssl mode.

My `.env` contains this line:
```
ROCKET_TLS='{certs="C:\Users\Administrator\Desktop\bitwarden-server\bitwarden_rs\bw-app\ssl\certificate.txt",key="C:\Users\Administrator\Desktop\bitwarden-server\bitwarden_rs\bw-app\ssl\key.txt"}'
```

I get this:

![image](https://user-images.githubusercontent.com/6625123/112680571-d5886f80-8e6d-11eb-8b04-0f4406cd93bf.png)

I think there is some convention i'm missing, since I also tried putting the files in the working dir and change the `.env` like this ...

```
ROCKET_TLS='{certs="cert.txt",key="key.txt"}'
```

**Comment by beppe9000** [comment]
I gave up in the end and used a [reverse proxy](https://github.com/dani-garcia/bitwarden_rs/wiki/Hardening-Guide#reverse-proxying) (Caddy), but I didn't give up on figuring what caused the problem.

**Comment by pepa65** [comment]


**Reply by beppe9000** [reply]
> Rocket runs inside the container

nope: I'm running the `bitwarden_rs.exe` executable I compiled from scratch on my windows machine without sandboxing
my understanding is that rocket is used as a library in there, to provide the web server capability to serve vault and api over http(s)

**Comment by BlackDex** [comment]
I think you need to use / instead of \ for this to work.

**Reply by beppe9000** [reply]
is it supposed to be an absolute or relative path? because I tried putting everything in the same folder and dropping the `\`s before and did not work.

**Reply by BlackDex** [reply]
Absolute or relative should both work, at least on linux.
Not sure how that works on windows.

**Reply by beppe9000** [reply]
So when i tried `ROCKET_TLS='{certs="cert.txt",key="key.txt"}'` (having the files in the CWD) it might've been a windows-specific problem ?

**Comment by pepa65** [comment]


**Reply by beppe9000** [reply]
the real problem is afaik windows lacks the concept of a root directory and instead uses multiple "volumes" that are mounted as separate absolute  paths. e.g. `c:\`, `d:\` and so on. in a way it is like having multiple roots, but `c:\..\` does not exist

**Comment by beppe9000** [comment]
i'll try to increase the log level, maybe the actual error will be printed

**Comment by pepa65** [comment]


**Reply by beppe9000** [reply]
the fact is i am not using docker at all... i am typing `bitwarden_rs.exe` in the windows command prompt
edit: no problem 😅

**Comment by beppe9000** [comment]
I guess I could also have changed the `Rocket.toml` file directly 🤔
